packages/client/src/index.ts

```diff
@@ -89,18 +89,41 @@ export class Client extends AutoGeneratedClient {
   }: CreateFileProps): Promise<CreateFileResponse> => {
     const headers = {
       ...this.config.headers,
-      'x-name': name,
-      'x-tags': tags ? JSON.stringify(tags) : undefined,
-      'x-index': index ? 'true' : 'false',
-      'x-access-policies': accessPolicies?.join(',') ?? undefined,
-      'content-type': contentType ?? false, // false ensures that axios does not use application/x-www-form-urlencoded if contentType is undefined
       'content-length': contentLength,
     }
 
-    const resp = await this.axiosClient.post('/v1/files', data, { headers, baseURL: this.config.apiUrl }).catch((e) => {
-      throw getError(e)
-    })
-    return resp.data as CreateFileResponse
+    const requestBody: CreateFileBody = {
+      name,
+      tags,
+      index,
+      accessPolicies,
+      contentType,
+    }
+
+    const response = await this.axiosClient
+      .post('/v1/files', requestBody, { headers, baseURL: this.config.apiUrl })
+      .catch((e) => {
+        throw getError(e)
+      })
+
+    const responseBody = response.data as CreateFileResponse
+
+    try {
+      await axios.put(responseBody.file.uploadUrl, data, {
+        maxBodyLength: Infinity,
+      })
+    } catch (err: any) {
+      throw new FileUploadError(`Failed to upload file: ${err.message}`, <AxiosError>err, responseBody.file)
+    }
+
+    return responseBody
```

Comment:
@spg Even though the method is named `createAndUploadFile` and the file has already been uploaded at this point, the file in the returned response object will still have the `uploadStatus` set to `PENDING`, which I think can be very confusing. Perhaps we should rename that field to `uploadProcessingStatus` to clarify some processing is actually involved on our end before a file is usable?

Also, something I just realized is that for some use-cases the delay from which the file is uploaded to the upload URL to the moment it's actually processed by the worker and it can be retrieved by the user using the `getFileContent` endpoint can be an annoyance or even a deal-breaker. For example, with the current design I don't think we can use the Files API for allowing users to send images through webchat as the webchat frontend would have to wait until the `uploadStatus` is changed to `COMPLETE` by the worker before it's able to make a separate API call to the `getFileContent` endpoint to obtain the file URL so it can be sent to the webchat integration and used by the bot?